multi-agent/adapters/outbound/mongo/blueprint_repository.py

```
@@ -26,9 +31,16 @@ def __init__(self,
         )
+        self._col.create_index(
+            [("identity.type", pymongo.ASCENDING),
+             ("identity.id", pymongo.ASCENDING),
+             ("spec_dict.name", pymongo.ASCENDING)],
+            unique=True,
+            background=True,
+        )
 
     def save(self, identity: Identity, spec: BlueprintDraft,
-             rid_refs: list[str], metadata: Dict[str, Any] = {}) -> str:
+             rid_refs: list[str], metadata: dict[str, Any] = {}) -> str:
```

Ruff (0.15.21)

[warning] Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)